base/range.jl

```diff
@@ -536,10 +498,25 @@ function getindex{T}(r::FloatRange{T}, i::Integer)
     convert(T, (r.start + (i-1)*r.step)/r.divisor)
 end
 
-function getindex{T}(r::LinSpace{T}, i::Integer)
+function getindex(r::LinSpace, i::Integer)
     @_inline_meta
     @boundscheck checkbounds(r, i)
-    convert(T, ((r.len-i)*r.start + (i-1)*r.stop)/r.divisor)
+    unsafe_getindex(r, i)
+end
+
+# This is separate to make it useful even when running with --check-bounds=yes
+function unsafe_getindex(r::LinSpace, i::Integer)
+    d = r.lendiv
+    j, a, b = ifelse(2i >= length(r), (i-1, r.start, r.stop), (length(r)-i, r.stop, r.start))
```

Comment:
I'm pretty sure you want the branch the other way, as you want `t = j/d < 0.5` (otherwise the inner fma buys you nothing, since `1-t` is exact).